### PR TITLE
Add support for cancelling executions

### PR DESCRIPTION
**Pull request by iMicknl**
Not sure if I should call it `cancel_command` or `cancel_execution`?

### CODE REVIEW

**Review by tetienne — APPROVED** [review]

**Comment by tetienne** [comment]
_cancel_command_ is fine IMO, as we _execute_command_

**Comment by vlebourl** [comment]
> _cancel_command_ is fine IMO, as we _execute_command_
fine by me as well

File: pyhoma/client.py [hunk 1]
```diff
@@ -196,6 +200,12 @@ async def __post(
             await self.check_response(response)
             return await response.json()
 
+    async def __delete(self, endpoint: str) -> Any:
+        """ Make a POST request to the TaHoma API """
```
Comment (vlebourl) [inline pyhoma/client.py:204]: Isn't that a DELETE request?

**Review by vlebourl — APPROVED** [review]

File: pyhoma/client.py (hunk 1 above)
Comment (tetienne) [reply pyhoma/client.py:204]: Good catch :D

File: pyhoma/client.py [hunk 2]
```diff
@@ -196,6 +200,12 @@ async def __post(
...
+        async with self.session.delete(f"{self.api_url}{endpoint}") as response:
+            await self.check_response(response)
+            return await response.json()
```
Comment (tetienne) [inline pyhoma/client.py:207]: The DELETE command does not return anything.

Comment (iMicknl) [reply pyhoma/client.py:207]: @tetienne but it can return a status code in the range of 400/500? That's why I kept the `check_response` check.

Comment (iMicknl) [reply pyhoma/client.py:207]: Sorry, I missed that you pointed to this line `return await response.json()`... DELETE responses can return a JSON response, however in this case they don't. Should I remove it or keep it for other (potential) use-cases.

Comment (tetienne) [reply pyhoma/client.py:207]: I suggest you to call raise_for_status: https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientResponse.raise_for_status and return nothing.

Comment (iMicknl) [reply pyhoma/client.py:207]: @tetienne but in that case we won't be able to throw exceptions like 'Unauthorized' etc? I would prefer to keep `await self.check_response(response)`, but I can remove `return await response.json()` for now?

Comment (tetienne) [reply pyhoma/client.py:207]: Indeed, remove `             return await response.json() ` will be enough.

**Pull request merged**